zimbra/src/main/java/fr/openent/zimbra/service/impl/MessageService.java

```diff
@@ -1060,7 +1065,7 @@ public void retrieveMailFromZimbra(JsonObject returnedMail, JsonObject to_user_i
             } else {
                 if (user.existsInZimbra()) {
                     // Etape 3 : On recherche en fonction de l'objet, de l'expéditeur, de la date et de la boite de réception le mail à supprimer
-                    String query = "* NOT in:\"" + "Sent" + "\"" + " AND subject:\"" + subject + "\"" + " AND from:\"" + from_mail + "\"" + " AND date:\"" + date + "\"";
+                    String query = "* NOT in:\"" + "Sent" + "\"" + " AND from:\"" + from_mail + "\"" + " AND msgid:\"" + msgid + "\"";
```

Comment:
As indicated in the Zimbra doc we can retrieve a mail with the header Message-ID : https://github.com/Zimbra/zd-full/blob/master/ZimbraServer/docs/query.txt#L5